Chaincase.UI/wwwroot/ionic/index.esm.js

```diff
@@ -0,0 +1 @@
+import"./p-53b5a7f5.js";export{g as getPlatforms,i as isPlatform}from"./p-275c7570.js";import"./p-a07772f1.js";export{c as createAnimation}from"./p-986d0fc8.js";export{a as LIFECYCLE_DID_ENTER,c as LIFECYCLE_DID_LEAVE,L as LIFECYCLE_WILL_ENTER,b as LIFECYCLE_WILL_LEAVE,d as LIFECYCLE_WILL_UNLOAD}from"./p-31a14cff.js";export{iosTransitionAnimation}from"./p-95a0a8a3.js";export{mdTransitionAnimation}from"./p-b52c4353.js";export{g as getTimeGivenProgression}from"./p-bac8255f.js";import"./p-0d9cab2d.js";export{createGesture}from"./p-a013b234.js";export{I as IonicSafeString}from"./p-1f87b1b9.js";import"./p-28891211.js";export{m as menuController}from"./p-669f6e03.js";export{b as actionSheetController,a as alertController,l as loadingController,m as modalController,p as pickerController,c as popoverController,t as toastController}from"./p-d8e5ca41.js";const o=o=>{const r=window,t=r.Ionic;if(!t||!t.config||"Object"===t.config.constructor.name)return r.Ionic=r.Ionic||{},r.Ionic.config=Object.assign(Object.assign({},r.Ionic.config),o),r.Ionic.config;console.error("ionic config was already initialized")},r=()=>{const o=window,r=o&&o.Ionic&&o.Ionic.config;return r?r.mode?r.mode:r.get("mode"):"md"};export{r as getMode,o as setupConfig}
```

Comment:
What's up with these literally hundreds of compiled js files? I think we should try to stay as strict as possible keeping this repo to source code. I know we can compile/minify these js & css files with MSBUILD or another build tool. I'm especially confused as to what all the p-<id>.entry.js files are. A project with this profile cannot have these unintelligible files as "source"

Reply:
These are the modules from the ionic dist folder from an npm module. The bunch of files means that modules are lazy loaded as they are used instead of delaying the initial app load

Reply:
Minifying and bundling this with the shitty ms tooling would definitely break this. I guess you can include a different dist build if you wanted to sacrifice some initial load time. Or maybe find a way to integrate npm into the csproj to add ionic as a dependency and selectively move the specific dist folder

Reply:
hmmm I'm deeply concerned with this method but perhaps not for a good reason. I say that cause the alternative is just to download the npm  package and transpile it down our own to distribution still trusting perhaps that ionic is behaving honestly. I think lazy load makes sense here if it can't be totally wasm but I don't understand enough about it

Reply:
check now. I added an npm package.json with the dependency. `npm i` is triggered before a build IF package.json has not changed. `npm i` also trigger an npm script `prepare` which copies over the ionic deps to the `wwwroot/ionic` folder

here's my working demo
https://github.com/Kukks/Blazor.Ionic/blob/master/Blazor.Ionic.Demo/Blazor.Ionic.Demo.csproj#L27
https://github.com/Kukks/Blazor.Ionic/blob/master/Blazor.Ionic.Demo/package.json